Tests/SonarScanner.MSBuild.Shim.Test/PathHelperTests.cs

```diff
@@ -29,164 +30,162 @@ namespace SonarScanner.MSBuild.Shim.Test
-                new DirectoryInfo("C:\\Foo\\Bar.cs"),
-                new DirectoryInfo("C:\\Foo\\FooBar.cs"),
-            }).FullName.Should().Be("C:\\");
-        }
+                new DirectoryInfo(@"C:\Temp"),
+                new DirectoryInfo(@"D:\ThreeTimes\A"),
+                new DirectoryInfo(@"D:\ThreeTimes\B"),
+                new DirectoryInfo(@"D:\ThreeTimes\C"),
+                new DirectoryInfo(@"E:\AlsoThreeTimes\A"),
+                new DirectoryInfo(@"E:\AlsoThreeTimes\B"),
+                new DirectoryInfo(@"E:\AlsoThreeTimes\C"),
+            }).Should().BeNull();
 
         [TestMethod]
-        public void GetParts_WhenNull_ThrowsArgumentNullException()
-        {
-            // Arrange
-            // Act
-            Action act = () => PathHelper.GetParts(null);
+        public void BestCommonPrefix_WhenNoCommonPath_ReturnsMostCommonOne_Simple() =>
+            PathHelper.BestCommonPrefix(new[]
+            {
+                new DirectoryInfo(@"C:\Temp"),
+                new DirectoryInfo(@"D:\WorkDir\Project"),
+                new DirectoryInfo(@"D:\WorkDir\Project.Tests"),
+            }).FullName.Should().Be(@"D:\WorkDir");
 
-            // Assert
-            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("directoryInfo");
-        }
+        [TestMethod]
+        public void BestCommonPrefix_WhenNoCommonPath_ReturnsMostCommonOne_Complex() =>
+            PathHelper.BestCommonPrefix(new[]
+            {
+                new DirectoryInfo(@"C:\Temp"),
+                new DirectoryInfo(@"D:\ThreeTimes\A"),
+                new DirectoryInfo(@"D:\ThreeTimes\B"),
+                new DirectoryInfo(@"D:\ThreeTimes\C"),
+                new DirectoryInfo(@"E:\Two\A"),
+                new DirectoryInfo(@"E:\Two\B"),
+            }).FullName.Should().Be(@"D:\ThreeTimes");
+
+        [TestMethod]
+        public void BestCommonPrefix_WhenCommonPath_ReturnsTheLongestCommonPart() =>
+            PathHelper.BestCommonPrefix(new[]
+            {
+                new DirectoryInfo(@"C:\Common"),
+                new DirectoryInfo(@"C:\Common\SubDirA"),
+                new DirectoryInfo(@"C:\Common\SomethingElse"),
+            }).FullName.Should().Be(@"C:\Common");
+
+        [TestMethod]
+        public void BestCommonPrefix_WhenCommonPathOfFiles_ReturnsTheLongestCommonPart() =>
+            PathHelper.BestCommonPrefix(new[]
+            {
+                new DirectoryInfo(@"C:\InRoot.cs"),
+                new DirectoryInfo(@"C:\SubDir\A.cs"),
+                new DirectoryInfo(@"C:\SubDir\B.cs"),
+            }).FullName.Should().Be(@"C:\");
+
+        [TestMethod]
+        public void GetParts_WhenNull_ThrowsArgumentNullException() =>
+            ((Action)(() => PathHelper.GetParts(null))).Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("directory");
 
         [TestMethod]
         public void GetParts_ReturnsTheExpectedValues()
```

Comment:
do we have unix tests for `GetParts`?

Reply:
I don't understand the question as it is placed on the answer. `GetParts` is tested via `GetParts_ReturnsTheExpectedValues` 😄 (and the other null corner case `GetParts_WhenNull_ThrowsArgumentNullException`)

Reply:
sorry - should `GetParts` support Unix paths, too? because I only see windows paths here.

Reply:
It supports. We can't properly test that as `DirectoryInfo` will convert everything to Windows paths before processing.

Reply:
Ah ok, I missed that point.